src/core/src/OmniTileset.cpp

```diff
@@ -509,6 +528,62 @@ void OmniTileset::addImageryIon(const pxr::SdfPath& imageryPath) {
     _imageryPaths.push_back(imageryPath);
 }
 
+void OmniTileset::addImageryPolygon(const pxr::SdfPath& imageryPath) {
+    const OmniPolygonImagery imagery(imageryPath);
+
+    const auto uniqueName = "imagery_polygon_test"; // DEVEL
+
+    auto polygonImagery = UsdUtil::getCesiumPolygonImagery(imageryPath);
+    auto cartographicPolygonsRel = polygonImagery.GetCartographicPolygonBindingRel();
+    pxr::SdfPathVector cartographicPolygonTargets;
+    cartographicPolygonsRel.GetTargets(&cartographicPolygonTargets);
+    std::vector<CesiumGeospatial::CartographicPolygon> polygons;
+    for (const auto& cartographicPolygonTarget : cartographicPolygonTargets) {
+        auto cartographicPolygon = UsdUtil::getCesiumCartographicPolygon(cartographicPolygonTarget);
+        auto basisCurves = UsdUtil::getUsdBasisCurves(cartographicPolygonTarget);
+        auto pointsAttr = basisCurves.GetPointsAttr();
+        pxr::VtArray<pxr::GfVec3f> points;
+        pointsAttr.Get(&points);
+        // Take every 3rd point starting at index 0. The other points are the control points off the curve of the spline.
+        std::vector<glm::dvec3> pointsOnCurve;
+        for (size_t i = 0; i < points.size(); i += 3) {
```

Comment:
This only works when the curve type is set to cubic.  When the curve is linear, it just ends up skipping vertices.  You can use this USDA to test both types side by side.

[twoTypeClippingTest.zip](https://github.com/CesiumGS/cesium-omniverse/files/13733755/twoTypeClippingTest.zip)

Set to i += 3
![image](https://github.com/CesiumGS/cesium-omniverse/assets/123468416/cf79bbad-34da-4b07-b985-e65338960302)

Set to i++
![image](https://github.com/CesiumGS/cesium-omniverse/assets/123468416/72ee7fe5-8d32-45c0-be63-554c9902ff90)

Annoyingly it's not as simple as asking the user to switch the type to linear, as this still retains the handle vertices in the list.  Either we just default to i++ and tell the user not to use bezier, or we make the iteration steps condition based on `basisCurves.GetTypeAttr()`

Reply:
@r-veenstra Great breakdown and test file. For this initial version, I like the idea of making the iteration steps based on `basisCurves.GetTypeAttr()`. It's more user friendly. But I'm not sure how to handle the case of someone switching from `bezier` to `linear`, since we'd still need loop with a stride of 3 due to retaining handle vertices in the list. We'd need a way to detect whether the curve was ever a `bezier` type. Given that, the only option I can see is telling users not to use `bezier`.

Ultimately, I think we should either create a new substitute for `BasicCurves` or find a way to constrain `BasisCurves` to only be linear. Once we can create these with the Fabric Scene Delegate enabled as a `CesiumCartographicPrim` instead of a converted `BasisCurves`, it should be very feasible.

Reply:
I think we can ignore `bezier `> `linear `conversion, just recommend to use linear for now and as you say revisit when direct authroing when FSD is enabled is feasible.